Makefile

```diff
@@ -5,18 +5,18 @@ OBJCOPY=arm-none-eabi-objcopy
 # Directories
 CMSIS_DIR=./CMSIS
 HAL_DIR=./hal
-SRC_DIR=./src
 INC_DIR=./inc
+SRC_DIR=./src
 
 # Flags
-CFLAG=-mcpu=cortex-m4 -mthumb -02 -Wall -I$(CMSIS_DIR) -I$(HAL_DIR) -I$(INC_DIR)
-LDFLAGS=-T $(SRC_DIR)/linker_script.ld -nostartfile
+CFLAG=-mcpu=cortex-m4 -mthumb -O2 -Wall -I$(CMSIS_DIR) -I$(HAL_DIR) -I$(INC_DIR)
```

Comment:
What the reason is to use -02 flag? does it need this optimization level?

Reply:
As the project is starting I can build it without optimization, but, I'm thinking that this could be a big project, so, I choose level 2 because it enhances performance without significantly increasing the code size or introducing the risks.